lib/mix/lib/mix/tasks/format.ex

```diff
@@ -112,6 +156,71 @@ defmodule Mix.Tasks.Format do
     end
   end
 
+  # This function reads exported configuration from the imported dependencies and deals with
+  # caching the result of reading such configuration in a manifest file.
+  defp fetch_deps_opts(formatter_opts) do
+    deps = formatter_opts[:import_deps]
+
+    cond do
+      is_nil(deps) or deps == [] ->
+        formatter_opts
+
+      is_list(deps) ->
+        dep_parenless_calls =
+          if deps_dot_formatters_stale?() do
+            dep_parenless_calls = eval_deps_opts(deps)
+
+            if dep_parenless_calls != [] do
+              write_deps_manifest(dep_parenless_calls)
+            end
+
+            dep_parenless_calls
+          else
+            read_deps_manifest()
+          end
+
+        Keyword.update(
+          formatter_opts,
+          :locals_without_parens,
+          dep_parenless_calls,
+          &(&1 ++ dep_parenless_calls)
+        )
+
+      true ->
+        Mix.raise("Expected :import_deps to return a list of dependencies, got: #{inspect(deps)}")
+    end
+  end
+
+  defp deps_dot_formatters_stale?() do
+    Mix.Utils.stale?([".formatter.exs" | Mix.Project.config_files()], [deps_manifest()])
+  end
+
+  defp deps_manifest() do
+    Path.join(Mix.Project.build_path(), ".cached_deps_formatter.exs")
+  end
+
+  defp read_deps_manifest() do
+    deps_manifest() |> File.read!() |> :erlang.binary_to_term()
+  end
+
+  defp write_deps_manifest(parenless_calls) do
+    manifest = deps_manifest()
+    File.mkdir_p!(Path.dirname(manifest))
+    File.write!(deps_manifest(), :erlang.term_to_binary(parenless_calls))
+  end
+
+  defp eval_deps_opts(deps) do
+    deps_paths = Mix.Project.deps_paths()
+
+    for dep <- deps,
+        dep_dot_formatter = Path.join(Map.fetch!(deps_paths, dep), ".formatter.exs"),
+        File.regular?(dep_dot_formatter),
+        {dep_opts, _} = Code.eval_file(dep_dot_formatter),
```

Comment:
What if dep_opts is not a keyword list?

Reply:
Pushed a fix that extracts reading a kw list from a file (using this when reading `.formatter.exs` as well).